signer/core/signed_data.go

```diff
@@ -0,0 +1,714 @@
+// Copyright 2018 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+//
+package core
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"mime"
+	"reflect"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/sha3"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type SigFormat struct {
+	Mime        string
+	ByteVersion byte
+}
+
+var (
+	TextValidator = SigFormat{
+		"text/validator",
+		0x00,
+	}
+	DataTyped = SigFormat{
+		"data/typed",
+		0x01,
+	}
+	ApplicationClique = SigFormat{
+		"application/clique",
+		0x02,
+	}
+	TextPlain = SigFormat{
+		"text/plain",
+		0x45,
+	}
+)
+
+type ValidatorData struct {
+	Address common.Address
+	Message hexutil.Bytes
+}
+
+type TypedData struct {
+	Types       EIP712Types  `json:"types"`
+	PrimaryType string       `json:"primaryType"`
+	Domain      EIP712Domain `json:"domain"`
+	Message     EIP712Data   `json:"message"`
+	Output      bytes.Buffer
+}
+
+type EIP712Type []map[string]string
+
+type EIP712Types map[string]EIP712Type
+
+type EIP712TypePriority struct {
+	Type  string
+	Value uint
+}
+
+type EIP712Data = map[string]interface{}
+
+type EIP712Domain struct {
```

Comment:
There are a lot of `EIP712XXX` types. I find them a bit unintuitive, and easy to confuse. Are all of these needed? Yes is an ok answer

Reply:
You're right, the naming indirectly requires the developer to go read more about EIP712. It should be more intuitive. Will change them to `TypedDataMessage` and `TypedDataDomain`.